### DISCUSSION TITLE
Having a Viash components repository

### DISCUSSION THREAD

**Discussion by Joaodemeirelles** [viash-io/viash#623] [Ideas]
Hello everyone!

Just learned about Viash and it seems a really cool tech!

Is there an idea of creating a Viash components repo/pool, such as nf-core has for modules? Example below:
https://github.com/nf-core/modules

It would be pretty cool I think to make it easy to create new pipelines/workflows and also learn how to use Viash itself to create components.

**Comment by tverbeiren** [comment]
Hi @Joaodemeirelles 

Thanks for trying out Viash, happy to read you like it!

Yes, we thought about creating a repository of components. As a matter of fact, we do not even _need_ a separate repository for this: every component in any Viash repo can be used as a dependency in another project.

An example of this can be found [in the a demo pipeline](https://viash-hub.com/data-intuitive/viash_hub_demo/-/blob/v0.1.0/src/workflows/parallel_qc/main.nf) that we recently created. Please note that that repository does not contain any components/modules of itself, everything is included as a dependency. 

As next steps, two of our current projects might be of interest to you:

1. In order to make it easier for people to both use and search more easily we are currently developing Viash Hub, a package manager for Viash modules and subworkflows.
2. We started transforming the [snakemake wrappers repository](https://github.com/snakemake/snakemake-wrappers/tree/master) into a Viash module repository, obviously to be hosted on Viash Hub.

Is there anything specific you would like to do or learn more about?

**Reply by Joaodemeirelles** [reply]
That is really great! The idea of Viash Hub actually is exactly what I had thought! 

Thank you so much for the answer.
